internal/chain/reward.go

```diff
@@ -45,34 +46,35 @@ func ballotResultBeaconchain(
 }
 
 var (
-	votingPowerCache   singleflight.Group
-	delegateShareCache singleflight.Group
+	votingPowerCache, _   = lru.New(16)
+	delegateShareCache, _ = lru.New(1024)
 )
 
 func lookupVotingPower(
 	epoch *big.Int, subComm *shard.Committee,
 ) (*votepower.Roster, error) {
-	key := fmt.Sprintf("%s-%d", epoch.String(), subComm.ShardID)
-	results, err, _ := votingPowerCache.Do(
-		key, func() (interface{}, error) {
-			votingPower, err := votepower.Compute(subComm, epoch)
-			if err != nil {
-				return nil, err
-			}
-
-			// For new calc, remove old data from 3 epochs ago
-			deleteEpoch := big.NewInt(0).Sub(epoch, big.NewInt(3))
-			deleteKey := fmt.Sprintf("%s-%d", deleteEpoch.String(), subComm.ShardID)
-			votingPowerCache.Forget(deleteKey)
+	// Look up
+	key := fmt.Sprintf("%s-%d", epoch.String(), subComm.Hash())
+	fmt.Println(key)
+	if b, ok := votingPowerCache.Get(key); ok {
+		if b == nil {
+			fmt.Println("DAMN")
```

Comment:
log this error please

Reply:
sorry, those are debug code. Forgot to remove. Removed.